include/revng/PipelineC/Prototypes.h

```diff
@@ -8,14 +8,25 @@
 
 /**
  * Every pointer returned by this library is owned by the library and must not
- * be deleted, except for pointers returned by functions containing the word
- * "create". Objects returned by functions named "create" are owned by the
- * caller and must be destroyed invoking the appropriate function called
- * rp_TYPE_destroy, or, in the case of strings, rp_string_destroy.
+ * be deleted, except for pointers returned by functions with the 'owning'
+ * comment in the return type, these are owned by caller and must be destroyed
+ * invoking the appropriate function called rp_TYPE_destroy, or, in the case of
+ * strings, rp_string_destroy.
  *
  * Unless otherwise noted, no pointer arguments can be NULL.
  */
 
+/**
+ * INVALIDATIONS
+ * If a user of the pipeline wishes to cache the output of
+ * rp_manager_produce_targets it can do so, however some modifications can lead
+ * to a produced output to be invalid. To help with this there is
+ * rp_invalidations, which if provided to the methods that have it in their
+ * argument, will fill in all the targets that have been invalidated by the
+ * operation. If the user is not interested in having invalidation information
+ * the parameter can be set to nullptr.
+ */
```

Comment:
Move this to the doxygen of one of the functions that has a `rp_invalidations *invalidations` argument and refer to it in the other functions.